tests/components/vacuum/test_xiaomi.py

```diff
@@ -0,0 +1,161 @@
+"""The tests for the Xiaomi vacuum platform."""
+import asyncio
+from datetime import timedelta
+from unittest import mock
+
+import pytest
+
+from homeassistant.components.vacuum import (
+    ATTR_BATTERY_ICON,
+    ATTR_FAN_SPEED, ATTR_FAN_SPEED_LIST, DOMAIN,
+    SERVICE_LOCATE, SERVICE_RETURN_TO_BASE, SERVICE_SEND_COMMAND,
+    SERVICE_SET_FAN_SPEED, SERVICE_STOP,
+    SERVICE_TOGGLE, SERVICE_TURN_OFF, SERVICE_TURN_ON)
+from homeassistant.components.vacuum.xiaomi import (
+    ATTR_CLEANED_AREA, ATTR_CLEANING_TIME, ATTR_DO_NOT_DISTURB, ATTR_ERROR,
+    CONF_HOST, CONF_NAME, CONF_TOKEN, PLATFORM,
+    SERVICE_MOVE_REMOTE_CONTROL, SERVICE_MOVE_REMOTE_CONTROL_STEP,
+    SERVICE_START_REMOTE_CONTROL, SERVICE_STOP_REMOTE_CONTROL)
+from homeassistant.const import (
+    ATTR_SUPPORTED_FEATURES, CONF_PLATFORM, STATE_OFF, STATE_ON)
+from homeassistant.setup import async_setup_component
+
+
+@pytest.fixture
+def mock_mirobo():
+    """Mock mock_mirobo."""
+    mock_vacuum = mock.MagicMock()
+    type(mock_vacuum.Vacuum().status()).data = {'test': 'raw'}
```

Comment:
I have very little experience with tests, mocking and so on in Python, that was 'the way' it works as I want, but, please, _enlighten me_.